storage/s3/s3.go

```diff
@@ -34,18 +36,55 @@ type minioStorage struct {
 }
 
 // New returns a new Storage that can store objects to S3.
-func New(bucket string, prefix string, metafilesPrefix string, mc MinioClient, l log.Logger) storage.Storage {
-	return &minioStorage{
-		bucket:          bucket,
-		mc:              mc,
-		l:               l,
-		prefix:          prefix,
-		metafilesPrefix: metafilesPrefix,
-		useDone:         metafilesPrefix != "",
+func New() storage.Storage {
+	return &minioStorage{}
+}
+
+const defaultEndpoint = "s3.amazonaws.com"
+
+type sourceConfig struct {
```

Comment:
This is duplicated with the `sourceConfig` type in plugins/s3. I definitely fear that one of them will become out of sync. If the `sourceConfig` isn't even used here, then it would be better not to declare this type an to directly put it in the `destinationConfig`

Reply:
I think we should only have one package for s3 that implements both the Destination and Source.

Maybe we could even consider to unify both interfaces and the plugins would just have to return the not implemented error if they only want to implement one interface.